test/extended/quota/resourcequota.go

```diff
@@ -216,6 +217,89 @@ var _ = g.Describe("[sig-api-machinery][Feature:ResourceQuota]", func() {
 			})
 			o.Expect(err).NotTo(o.HaveOccurred())
 		})
+
+		g.It("when exceed openshift.io/image-tags will ban to create new image references in the project [Skipped:Disconnected]", func() {
+			testProject := oc.SetupProject()
+			testResourceQuotaName := "my-image-tag-quota-" + testProject
+			clusterAdminKubeClient := oc.AdminKubeClient()
+
+			lr := &corev1.LimitRange{
+				ObjectMeta: metav1.ObjectMeta{Name: testResourceQuotaName, Namespace: testProject},
+				Spec: corev1.LimitRangeSpec{
+					Limits: []corev1.LimitRangeItem{
+						{
+							Type: "openshift.io/ImageStream",
+							Max: corev1.ResourceList{
+								"openshift.io/image-tags": resource.MustParse("2"),
+							},
+						},
+					},
+				},
+			}
+
+			g.By("create the image-tags and checking the usage")
+			_, err := clusterAdminKubeClient.CoreV1().LimitRanges(testProject).Create(context.Background(), lr, metav1.CreateOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			err = waitForLimitRangeStatus(clusterAdminKubeClient, testResourceQuotaName, testProject, func(actualLimitRange *corev1.LimitRange) error {
+				expectedLimitRange := []corev1.LimitRangeItem{
+					{
+						Type: "openshift.io/ImageStream",
+						Max: corev1.ResourceList{
+							"openshift.io/image-tags": resource.MustParse("2"),
+						},
+					},
+				}
+				// Sorting actual and expected LimitRangeItems for comparison
+				sort.Slice(actualLimitRange.Spec.Limits, func(i, j int) bool {
+					return actualLimitRange.Spec.Limits[i].Type < actualLimitRange.Spec.Limits[j].Type
+				})
+				sort.Slice(expectedLimitRange, func(i, j int) bool {
+					return expectedLimitRange[i].Type < expectedLimitRange[j].Type
+				})
+
+				// Compare actual and expected LimitRangeItems
+				if !equality.Semantic.DeepEqual(actualLimitRange.Spec.Limits, expectedLimitRange) {
+					return fmt.Errorf("unexpected current total usage: actual: %#v, expected: %#v", actualLimitRange.Spec.Limits, expectedLimitRange)
+				}
+				actualStr := fmt.Sprintf("%v", actualLimitRange.Spec.Limits)
+				expectedStr := fmt.Sprintf("%v", expectedLimitRange)
+				fmt.Printf("Actual: %s\nExpected: %s\n", actualStr, expectedStr)
+				return nil
+			})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			images := []struct {
+				Image string
+				Tag   string
+			}{
+				{
+					Image: "quay.io/openshifttest/hello-openshift@sha256:4200f438cf2e9446f6bcff9d67ceea1f69ed07a2f83363b7fb52529f7ddd8a83",
+					Tag:   "v1",
+				},
+				{
+					Image: "quay.io/openshifttest/base-alpine@sha256:3126e4eed4a3ebd8bf972b2453fa838200988ee07c01b2251e3ea47e4b1f245c",
+					Tag:   "v2",
+				},
+				{
+					Image: "openshift/hello-openshift",
+					Tag:   "v3",
+				},
+			}
+
+			for _, u := range images {
+				g.By("trying to tag a container image with " + u.Tag)
+				err = oc.AsAdmin().WithoutNamespace().Run("tag").Args(u.Image, "--source=docker", "mystream:"+u.Tag, "-n", testProject).Execute()
```

Comment:
This is wrong, you should run this as a regular user, iow.: 
```golang
err = oc.Run("tag").Args(u.Image, "--source=docker", "mystream:"+u.Tag, "-n", testProject).Execute()
```
since you don't need to be admin, and you're tagging an image stream which is namespaced.

Reply:
Updated